src/inference/Inference.jl

```diff
@@ -317,7 +317,8 @@ function AbstractMCMC.bundle_samples(
     model::AbstractModel,
     spl::Sampler,
     N::Integer,
-    ts::Vector{<:AbstractTransition};
+    ts::Vector{<:AbstractTransition},
+    chain_type::Type{Any};
```

Comment:
IMO the more intuitive approach would be to use
```suggestion
    chain_type::Type{Chains};
```
here, since this implementation returns a `Chains` object. If this is supposed to be the default behaviour in Turing, I think one should just set these defaults in the Turing implementations of `sample`, i.e., one should use something similar to
```julia
sample([rng::AbstractRNG, ]model::AbstractModel, alg::InferenceAlgorithm, N::Integer; kwargs...) =
    sample(rng, model, Sampler(alg, model), N; chain_type = Chains, kwargs...)
```
Then there would be also no need for defining the method for `chain_type = Vector`, since one could just use `sample(rng, model, alg, N; chain_type = Any)` to get the default behaviour of returning just the set of all transitions.